pulseapi/profiles/views.py

```diff
@@ -66,3 +67,62 @@ def put(self, request, *args, **kwargs):
             pass
 
         return super(UserProfileAPIView, self).put(request, *args, **kwargs)
+
+
+class ProfileCustomFilter(filters.FilterSet):
+    """
+      We add custom filtering to allow you to filter by:
+
+      * Profile type - pass the `?profile_type=` query parameter
+      * Program type - pass the `?program_type=` query parameter
+      * Program year - pass the `?program_year=` query parameter
+    """
+    profile_type = django_filters.CharFilter(
+        name='profile_type__value',
+        lookup_expr='iexact',
+    )
+    program_type = django_filters.CharFilter(
+        name='program_type__value',
+        lookup_expr='iexact',
+    )
+    program_year = django_filters.CharFilter(
+        name='program_year__value',
+        lookup_expr='iexact',
+    )
+
+    @property
+    def qs(self):
```

Comment:
let's add a test for an invalid query param

Reply:
added.